pkg/kp/rcstore/consul_store.go

```diff
@@ -138,31 +138,24 @@ func (s *consulStore) List() ([]fields.RC, error) {
 func (s *consulStore) WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error) {
 	outCh := make(chan []fields.RC)
 	errCh := make(chan error)
+	inCh := make(chan api.KVPairs)
+
+	go consulutil.WatchPrefix(kp.RC_TREE, s.kv, inCh, quit, errCh)
 
 	go func() {
 		defer close(outCh)
 		defer close(errCh)
-		var currentIndex uint64 = 0
 
 		for {
 			select {
 			case <-quit:
 				return
-			case <-time.After(1 * time.Second):
-				listed, meta, err := s.kv.List(kp.RC_TREE, &api.QueryOptions{
-					WaitIndex: currentIndex,
-				})
+			case listed := <-inCh:
```

Comment:
You could simplify this to a `for listed := range inCh {` loop.  It would additionally fix the ordering of writes to the error channel.